index.js

```diff
@@ -265,18 +269,36 @@ FacebookInsightStream.prototype._collect = function ( metrics, item, buffer, eve
                 }
             }
         })
+        .catch( SkipedError, function ( error ) {
+            console.log( "facebook-insights skiped error", error );    
+        })
         .then( function () {
+            // remove the current paramater when done
+            var _metricIdx = metrics.indexOf( _metric );
+            var _evIdx = events.indexOf( _ev );
+            metrics.splice( _metricIdx, 1 );
+            events.splice( _evIdx, 1 );
+
             return this._collect( metrics, item, buffer, events );
         })
         .catch( function ( error ) {
-            if ( error.skip ) {
-                return this._collect( metrics, item, buffer )
-            } else {
-                this.emit( "error", error )
-            }
+            var retry = this._collect.bind( this, metrics, item, buffer, events );
+            return this.handleError( error, retry );
```

Comment:
When there was an error but the hanldeError decide to retry and not to emit error, i will add some documentation